drivers/wifi/nrf_wifi/src/net_if.c

```diff
@@ -371,25 +371,25 @@ int nrf_wifi_if_send(const struct device *dev,
 		goto out;
 	}
 
+	/* Allocate packet before locking mutex (blocks until allocation sucess) */
+	nbuf = net_pkt_to_nbuf(pkt);
+	__ASSERT_NO_MSG(nbuf != NULL);
```

Comment:
Please reinstate `host_stats->total_tx_drop_pkts++;`

Reply:
My understanding is that it is not possible.
At this point we no longer hold the `vif_lock`, and we haven't gone through the error checking paths to get the `host_stats` pointer.
Is it even possible for this function to fail anymore? The function blocks until it succeeds.

Reply:
If can fail in case of mis-configured driver DATA_HEAP, and we memory allocation failures.

Reply:
> At this point we no longer hold the vif_lock, and we haven't gone through the error checking paths to get the host_stats pointer.

Yeah, but this statistic is important to see if driver is dropping packets.

Reply:
> If can fail in case of mis-configured driver DATA_HEAP, and we memory allocation failures.

Can it? The only paths I can see for failure are chronic misconfigurations that will always fail, not just sometimes:
* `size_t` overflows
* allocating more memory than the pool contains

These errors seem better served with an assertion or a `LOG_ERR`, not a counter that requires extra logic to look at.

Reply:
> allocating more memory than the pool contains

This is what I was referring to.

> These errors seem better served with an assertion or a LOG_ERR, not a counter that requires extra logic to look at.

Sure, the existing code had both LOG_ERR and statistics. Most of our customers don't enable UART in the field, so, statistics are the only way to know driver failures.